ui.apps/src/main/content/jcr_root/apps/venia/components/commerce/searchbar/clientlibs/js/searchbar.js

```diff
@@ -0,0 +1,247 @@
+  _init() {
+    this._initLiveSearch();
+  }
+
+  _injectStoreScript(src) {
+    const script = document.createElement("script");
+    script.type = "text/javascript";
+    script.src = src;
+
+    document.head.appendChild(script);
+  }
+
+  async _getStoreData() {
+    const { dataServicesStorefrontInstanceContext, storeConfig } =
+      (await getGraphQLQuery(dataServicesStorefrontInstanceContextQuery)) || {};
+    this._state.dataServicesStorefrontInstanceContext =
+      dataServicesStorefrontInstanceContext;
+    this._state.storeConfig = storeConfig;
+    if (!dataServicesStorefrontInstanceContext) {
+      console.log("no dataServicesStorefrontInstanceContext");
+      return;
+    }
+  }
+  async _getMagentoExtensionVersion() {
+    const { dataServicesMagentoExtensionContext } =
+      (await getGraphQLQuery(dataServicesMagentoExtensionContextQuery)) || {};
+    this._state.magentoExtensionVersion =
+      dataServicesMagentoExtensionContext?.magento_extension_version;
+    if (!dataServicesMagentoExtensionContext) {
+      console.log("no magentoExtensionVersion");
+      return;
+    }
+  }
+
+  async _initLiveSearch() {
+    await Promise.all([
+      this._getStoreData(),
+      this._getMagentoExtensionVersion(),
+    ]);
+    if (!window.LiveSearchAutocomplete) {
+      this._injectStoreScript(
+        "https://searchautocompleteqa.magento-datasolutions.com/v0/LiveSearchAutocomplete.js"
+      );
+      // wait until script is loaded
+      await new Promise((resolve) => {
+        const interval = setInterval(() => {
+          if (window.LiveSearchAutocomplete) {
+            clearInterval(interval);
+            resolve();
+          }
+        }, 200);
+      });
+    }
+    const { dataServicesStorefrontInstanceContext } = this._state;
+    if (!dataServicesStorefrontInstanceContext) {
+      console.log("no dataServicesStorefrontInstanceContext");
+      return;
+    }
+    // initialize live-search
+    new window.LiveSearchAutocomplete({
+      environmentId: dataServicesStorefrontInstanceContext.environment_id,
+      websiteCode: dataServicesStorefrontInstanceContext.website_code,
+      storeCode: dataServicesStorefrontInstanceContext.store_code,
+      storeViewCode: dataServicesStorefrontInstanceContext.store_view_code,
+      config: {
+        pageSize: 8,
+        minQueryLength: "2",
+        currencySymbol: "$",
+        currencyRate: "1",
+        displayOutOfStock: true,
+        allowAllProducts: false,
+      },
+      context: {
+        customerGroup: dataServicesStorefrontInstanceContext.customer_group,
+      },
+    });
+
+    const formEle = document.getElementById("search_mini_form");
+
+    formEle.setAttribute(
+      "action",
+      `${dataServicesStorefrontInstanceContext.store_url}catalogsearch/result`
+    );
+    // initialize store event after live-search
+    this._initStoreEvent();
+  }
+
+  async _initStoreEvent() {
+    //  Magento Store event
+    if (!window.magentoStorefrontEvents) {
```

Comment:
You should not need this. CIF already embeds the storefront-events-sdk and collector in the `venia.cif` clientlib https://github.com/adobe/aem-cif-guides-venia/blob/c191e60560b45d840615f4925003a2e1f47c1d66/ui.apps/src/main/content/jcr_root/apps/venia/clientlibs/clientlib-cif/.content.xml#L8

Not sure if this is the latest version though. However, we should not load it twice so if the one in CIF is outdated we may want to try to update it. There is also another way to get access to the storefront events sdk via the react frontend we use in venia (based on PWA studio). We could expose it there and provide it as Promise on `window`.

Reply:
no, just the `if (!window.magentoStorefrontEvents) { ... }`.

If it works the sdk should already be available on `window` and the promise below will resolve immediately.